packages/fxa-admin-server/src/auth/user-group-header.guard.ts

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { GqlExecutionContext } from '@nestjs/graphql';
+import { USER_GROUP_HEADER, guard, AdminPanelFeature } from 'fxa-shared/guards';
+import { FEATURE_KEY } from './user-group-header.decorator';
+
+@Injectable()
+export class UserGroupGuard implements CanActivate {
+  constructor(private reflector?: Reflector) {}
+
+  canActivate(context: ExecutionContext): boolean {
+    // Reflect on the end point to determine if it has been tagged with admin panel feature.
+    // If it does, check to make sure the user's group has a permission level that permits access.
+    const features =
+      this.reflector?.getAllAndOverride<AdminPanelFeature[]>(FEATURE_KEY, [
+        context.getHandler(),
+        context.getClass(),
+      ]) || [];
+
+    if (!features || features.length === 0) {
```

Comment:
What is `!features` doin'?
Does that imply that `features` can be null/undefined or some other falsy value, or is it always an array based on the `|| []` bit?

Reply:
Same question as Peter.

Reply:
Ah good catch. I added the || [] after the fact and missed the next line.